pulser-core/pulser/sequence.py

```diff
@@ -547,6 +545,17 @@ def declare_channel(
             else:
                 raise ValueError(f"Channel {channel_id} is not available.")
 
+        if initial_target is not None:
+            try:
+                cond = any(
+                    isinstance(t, Parametrized)
+                    for t in cast(Iterable, initial_target)
+                )
+            except TypeError:
+                cond = isinstance(initial_target, Parametrized)
```

Comment:
Shouldn't we raise this TypeError if `cond == False` ? I think something like this would work:
```suggestion
            except TypeError as e:
                cond = isinstance(initial_target, Parametrized)
                if not cond:
                    raise e
```

Reply:
I was thinking of the case where you catch a different `TypeError` that doesn't come from the initial target and just ignore it

Reply:
Discussed together.